src/main/java/com/google/codeu/servlets/UserFormServlet.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Tea;
+
+
+
+
+public class UserFormServlet extends HttpServlet {
+  
+  private Datastore datastore;
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+    datastore = createDatastore();
+  }
+
+  /*
+   * Instantiates a connection to a datastore. In tests, override this with a
+   * method that returns a fake/mock datastore object.
+   */
+  protected Datastore createDatastore() {
+    return new Datastore();
+  }
+
+@Override
+public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+  UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect("/index.html");
+      return;
+    }
+    String user = userService.getCurrentUser().getEmail();
+    int greenTea = Integer.parseInt(request.getParameter("greenTea"));
```

Comment:
What happens if the parsing fails? It looks like this will result in a `500` (internal server error) status code being generated due to an uncaught exception. However, the appropriate response should be a `400` (bad request).

I would suggest doing this parsing with some error-handling logic that will emit an appropriate `400` status code (with content that indicates the part of the request that failed to validate correctly).